### PR TITLE
Allow use with Carthage

### PR DESCRIPTION
**Pull request by jkennington**
I wanted to be able to use this with Carthage. So frameworks targets have been added for macOS, iOS, tvOS, and watchOS.
I needed topological sorting and cycle detection. Those have been added here.
The demo app was moved into a sample project and a playground was added. 

 I don't use CocoaPods so I have not tested that with these changes.

 I'd like to remove the the free functions and work them into extensions soon you can just call `someGraph.dijkstra(...` instead of `dijkstra(graph: someGraph...`. I don't know if you are opposed to that. If you like the idea, I'm happy to submit more pull requests later.

### CODE REVIEW

**Comment by jkennington** [comment]
Also, if you wish to keep the frameworks but lose the cycle check and topological sort, just remove the two files with the extensions. I should have done those in separate pull requests.

**Comment by davecom** [comment]
Hi @jkennington,

First, thank you for the contributions. I really do appreciate your thought and work here. I think we really have three separate issues:

1. Adding support for detecting acyclic graphs and topological sort. I completely support this, but we need accompanying unit tests to go with it. Could you provide just the acyclic graph detection and the topological sort along with unit tests as a separate pull request?

2. Refactoring the search methods (bfs, dfs, dijkstra) to be extensions on `Graph` and `WeightedGraph`. I'm not opposed to this although it would break backwards compatibility with some projects I know of that are using SwiftGraph. Another issue is that some of the utility methods that work with them would still be free functions(`pathDictToPath()`, `distanceArrayToVertexDict()`). I can work on this for the next breaking release (1.2.0).

3. Restructuring the project to support Carthage. While I'm not opposed to Carthage per se, I don't use it myself so I can't maintain this and I don't want to rely on anyone else to maintain SwiftGraph. CocoaPods support is very important because several other projects are already incorporating SwiftGraph via CocoaPods. I also like the current project structure in terms of how easy it is for newcomers to just open the project and click run and see the demo app. SwiftGraph support SPM which I believe is the future of Swift package management. So, I'm not against including Carthage support but it needs some comprehensive support structure behind it and testing with existing CocoaPods and SPM integration. Therefore I would like to do it slowly and completely separated from everything else. I would rather do the other two items here first.

In short - I think this is all good, but we need to break it up into several separate pull requests. We also need unit tests for the new functionality and everything from here on out must be Apache licensed.

Thanks,
Dave

**Comment by jkennington** [comment]
Makes sense to me. I will work on this on my fork until stuff is in a better state and able to be test with SPM and Cocoapods. I think the other free functions you mentions could become class methods on `Graph` or `WeightedGraph` as appropriate. I'll plan to work towards a well tested but new version since there would be breaking changes.

I'll look into a better way to have the sample app and framework code together in a way that works well for SPM, Carthage, and Cocoapods.

**Comment by davecom** [comment]
@jkennington, thanks for understanding. I've refactored the search methods and dijkstra into an extension in the latest commit. Hope this helps.

**Comment by davecom** [comment]
@jkennington, I wrote and added implementations of topologicalSort() and isDAG() to an extension in the base repository. Checkout the latest commit and let me know if these are working for you. Thanks!
